app/src/main/java/tech/ula/ui/AppListFragment.kt

```diff
@@ -179,4 +201,25 @@ class AppListFragment : Fragment() {
     private fun passDataToActivity(data: String) {
         dataPasser.onFragmentDataPassed(data)
     }
-}
+
+    private fun selectServiceTypePreference(selectedApp: App) {
+        lateinit var dialog: AlertDialog
+
+        val serviceTypes = arrayOf("SSH", "VNC")
+        var preferredServiceType = "SSH"
+
+        val builder = AlertDialog.Builder(activityContext)
+                .setTitle("Always open with:")
+                .setSingleChoiceItems(serviceTypes, 0) { _, selected ->
+                    preferredServiceType = serviceTypes[selected]
+                }
+
+        builder.setPositiveButton("Continue") { _, _ ->
+            appListViewModel.setAppServiceTypePreference(selectedApp, preferredServiceType)
+            Toast.makeText(activityContext, "$preferredServiceType selected", Toast.LENGTH_SHORT).show()
```

Comment:
User facing text needs to be defined in `strings.xml`, but this could also just be removed imo.